In [155]:
# Imports
import pandas as pd
import os
import numpy as np

In [156]:
# Load CSV files in data/raw and read them as dataframes
data_dir = '../../data/raw'
csv_files = [f for f in os.listdir(data_dir) if f.endswith('.csv')]

dataframes = {}
for file in csv_files:
    file_path = os.path.join(data_dir, file)
    df_name = os.path.splitext(file)[0]
    dataframes[df_name] = pd.read_csv(file_path)

# Display the loaded dataframes
for name, df in dataframes.items():
    print(f"DataFrame: {name}")
    display(df)

DataFrame: dim_physician


PHYSICIAN_ID STATE             PHYSICIAN_TYPE GENDER  BIRTH_YEAR
0                 1    TX           PHYSICAL THERAPY    NaN         NaN
1                 2    IN        PHYSICIAN ASSISTANT    NaN         NaN
2                 3    CA         EMERGENCY MEDICINE    NaN         NaN
3                 4    TX         NURSE PRACTITIONER    NaN         NaN
4                 5    WA         EMERGENCY MEDICINE    NaN         NaN
...             ...   ...                        ...    ...         ...
25338         12730    FL            FAMILY MEDICINE      F      1981.0
25339         12731    FL  INTERVENTIONAL CARDIOLOGY      M      1975.0
25340         12732    CA              OPHTHALMOLOGY      M      1973.0
25341         12733    MI            GENERAL SURGERY      M      1975.0
25342         12734    FL         EMERGENCY MEDICINE      F      1980.0

[25343 rows x 5 columns]

DataFrame: Data Dictionary


Unnamed: 0     Unnamed: 1         Unnamed: 2 Unnamed: 3  \
0          NaN            NaN                NaN        NaN   
1          NaN     Table Name        Column Name  Data Type   
2          NaN       fact_txn             TXN_DT       DATE   
3          NaN       fact_txn         PATIENT_ID     NUMBER   
4          NaN       fact_txn       PHYSICIAN_ID     NUMBER   
5          NaN       fact_txn  TXN_LOCATION_TYPE     STRING   
6          NaN       fact_txn     INSURANCE_TYPE     STRING   
7          NaN       fact_txn           TXN_TYPE     STRING   
8          NaN       fact_txn           TXN_DESC     STRING   
9          NaN    dim_patient         PATIENT_ID     NUMBER   
10         NaN    dim_patient         BIRTH_YEAR     NUMBER   
11         NaN    dim_patient             GENDER     STRING   
12         NaN  dim_physician       PHYSICIAN_ID     NUMBER   
13         NaN  dim_physician              STATE     STRING   
14         NaN  dim_physician     PHYSICIAN_TYPE     STRING   
15         NaN  dim_physician             GENDER     STRING   
16         NaN  dim_physician         BIRTH_YEAR     NUMBER   

                                           Unnamed: 4  \
0                                                 NaN   
1                                       Values/Format   
2                                          YYYY-MM-DD   
3                                                 NaN   
4                                                 NaN   
5                                                 NaN   
6                                                 NaN   
7   CONDITIONS\nSYMPTOMS\nCONTRAINDICATIONS\nTREAT...   
8                                                 NaN   
9                                                 NaN   
10                                               YYYY   
11                        M-Male\nF-Female\nU-Unknown   
12                                                NaN   
13                                                NaN   
14                                                NaN   
15                                   M-Male\nF-Female   
16                                               YYYY   

                                           Unnamed: 5  
0                                                 NaN  
1                                        Descriptions  
2                         The date of the transaction  
3   Unique identifier for each patient (use to joi...  
4   Unique identifier for each physician associate...  
5         Type of location where transaction occurred  
6   Payment type classification as COMMERCIAL, MED...  
7   Identifies type of information identified duri...  
8       Short description of the recorded transaction  
9   Unique identifier for each patient (use to joi...  
10                           Patient's year of birth.  
11                                   Patient's Gender  
12  Unique identifier for each physician associate...  
13                    Physician's primary state code.  
14            Physician primary specialty description  
15                                 Physician's Gender  
16                         Physician's year of birth.

DataFrame: model_table


Column Name Data Type  \
0       PATIENT_ID    NUMBER   
1           TARGET    NUMBER   
2      DISEASEX_DT      DATE   
3      PATIENT_AGE    NUMBER   
4   PATIENT_GENDER    STRING   
5   NUM_CONDITIONS    NUMBER   
6   PHYSICIAN_TYPE    STRING   
7  PHYSICIAN_STATE    STRING   
8    LOCATION_TYPE    STRING   

                                       Values/Format  \
0                                                NaN   
1  1: Patient received Drug A\n0: Patient did not...   
2                                         YYYY-MM-DD   
3                                                NaN   
4                                   M-Male\nF-Female   
5                                                NaN   
6                                                NaN   
7                                                NaN   
8                                                NaN   

                                        Descriptions  
0                 Unique identifier for each patient  
1  Drug A indicator, used as target variable for ...  
2          Date patient was diagnosed with Disease X  
3       Age of patient when diagnosed with Disease X  
4                                   Patient's Gender  
5  Number of conditions making a patient higher r...  
6  Primary specialty description of physician dia...  
7            State of physician diagnosing Disease X  
8  Type of location where patient was diagnosed w...

DataFrame: fact_txn


TXN_DT  PATIENT_ID  PHYSICIAN_ID          TXN_LOCATION_TYPE  \
0       2021-12-27        3124           NaN     INDEPENDENT LABORATORY   
1       2020-02-25        3352       21972.0                     OFFICE   
2       2022-01-25        2902       21586.0        HOSPITAL OUTPATIENT   
3       2022-03-05        2943        2439.0                 UNASSIGNED   
4       2018-01-03        3589        6381.0                     OFFICE   
...            ...         ...           ...                        ...   
115269  2018-01-17        3208           NaN     INDEPENDENT LABORATORY   
115270  2017-02-26        3159           NaN           AMBULANCE - LAND   
115271  2017-01-19        2861        5661.0                     OFFICE   
115272  2021-12-05        3067       12133.0        HOSPITAL OUTPATIENT   
115273  2022-06-13        2945        2864.0  EMERGENCY ROOM - HOSPITAL   

       INSURANCE_TYPE    TXN_TYPE           TXN_DESC  
0          COMMERCIAL    SYMPTOMS  ACUTE_PHARYNGITIS  
1          COMMERCIAL  CONDITIONS            obesity  
2            MEDICARE  CONDITIONS  immunocompromised  
3          COMMERCIAL  CONDITIONS           diabetes  
4            MEDICARE    SYMPTOMS              COUGH  
...               ...         ...                ...  
115269       MEDICARE  CONDITIONS      heart_disease  
115270       MEDICARE    SYMPTOMS            FATIGUE  
115271       MEDICARE  CONDITIONS           diabetes  
115272     COMMERCIAL    SYMPTOMS              cough  
115273       MEDICARE  CONDITIONS          disease_x  

[115274 rows x 7 columns]

DataFrame: dim_patient


PATIENT_ID  BIRTH_YEAR GENDER
0              1        1988      M
1              2        2020      M
2              3        1973      M
3              4        2022      M
4              5        1988      M
...          ...         ...    ...
4015        4016        1947      F
4016        4017        1937      F
4017        4018        1937      F
4018        4019        1958      M
4019        4020        1941      F

[4020 rows x 3 columns]

In [157]:
# Clean data dictionary
dataframes['data_dic'] = dataframes.pop('Data Dictionary')
data_dic = dataframes['data_dic']
data_dic.rename(columns=data_dic.iloc[1], inplace = True)
data_dic = data_dic.iloc[2:, 1:]
data_dic

Table Name        Column Name Data Type  \
2        fact_txn             TXN_DT      DATE   
3        fact_txn         PATIENT_ID    NUMBER   
4        fact_txn       PHYSICIAN_ID    NUMBER   
5        fact_txn  TXN_LOCATION_TYPE    STRING   
6        fact_txn     INSURANCE_TYPE    STRING   
7        fact_txn           TXN_TYPE    STRING   
8        fact_txn           TXN_DESC    STRING   
9     dim_patient         PATIENT_ID    NUMBER   
10    dim_patient         BIRTH_YEAR    NUMBER   
11    dim_patient             GENDER    STRING   
12  dim_physician       PHYSICIAN_ID    NUMBER   
13  dim_physician              STATE    STRING   
14  dim_physician     PHYSICIAN_TYPE    STRING   
15  dim_physician             GENDER    STRING   
16  dim_physician         BIRTH_YEAR    NUMBER   

                                        Values/Format  \
2                                          YYYY-MM-DD   
3                                                 NaN   
4                                                 NaN   
5                                                 NaN   
6                                                 NaN   
7   CONDITIONS\nSYMPTOMS\nCONTRAINDICATIONS\nTREAT...   
8                                                 NaN   
9                                                 NaN   
10                                               YYYY   
11                        M-Male\nF-Female\nU-Unknown   
12                                                NaN   
13                                                NaN   
14                                                NaN   
15                                   M-Male\nF-Female   
16                                               YYYY   

                                         Descriptions  
2                         The date of the transaction  
3   Unique identifier for each patient (use to joi...  
4   Unique identifier for each physician associate...  
5         Type of location where transaction occurred  
6   Payment type classification as COMMERCIAL, MED...  
7   Identifies type of information identified duri...  
8       Short description of the recorded transaction  
9   Unique identifier for each patient (use to joi...  
10                           Patient's year of birth.  
11                                   Patient's Gender  
12  Unique identifier for each physician associate...  
13                    Physician's primary state code.  
14            Physician primary specialty description  
15                                 Physician's Gender  
16                         Physician's year of birth.

In [158]:
# Review dataframes info
for name, df in dataframes.items():
    print(f"DataFrame: {name}")
    df.info()
    print("\n")

DataFrame: dim_physician
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 25343 entries, 0 to 25342
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   PHYSICIAN_ID    25343 non-null  int64  
 1   STATE           25341 non-null  object 
 2   PHYSICIAN_TYPE  25343 non-null  object 
 3   GENDER          24137 non-null  object 
 4   BIRTH_YEAR      21039 non-null  float64
dtypes: float64(1), int64(1), object(3)
memory usage: 990.1+ KB


DataFrame: model_table
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9 entries, 0 to 8
Data columns (total 4 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   Column Name    9 non-null      object
 1   Data Type      9 non-null      object
 2   Values/Format  3 non-null      object
 3   Descriptions   9 non-null      object
dtypes: object(4)
memory usage: 420.0+ bytes


DataFrame: fact_txn
<class 'pandas.core.frame.DataFrame'>


Notes:
- The `fact_txn` dataframe has 115,274 rows and 7 columns, containing transaction data.
- - physician_id and txn_location_type have null values, some datatypes shall be changed
- The `dim_patient` dataframe has 4,020 rows and 3 columns, containing patient demographic information.
- - non null values, requires datatypes changes 
- The `dim_physician` dataframe has 25,343 rows and 5 columns, containing physician demographic information.
- - gender and birth_year have some null values, datatypes changes
- The `model_table` dataframe has 2 columns, containing model-related data.
- The `data_dic` dataframe provides metadata for the other dataframes, including column names, data types, and descriptions.

In [159]:
# Clean and transform fact_txn dataframe
fact_txn = dataframes['fact_txn']

# Handle missing values
#fact_txn['PHYSICIAN_ID'].fillna(-1, inplace=True)  # Fill missing physician_id with -1
#fact_txn.fillna({'TXN_LOCATION_TYPE': 'Unknown'}, inplace=True) # Fill missing txn_location_type with 'Unknown'

# Convert data types
# Datetime
fact_txn['TXN_DT'] = pd.to_datetime(fact_txn['TXN_DT'])  # Convert TXN_DT to datetime
# Numeric
#errors='coerce' invalid values will be set as NaN, Int64 will convert NaN to NA pandas
fact_txn['PATIENT_ID'] = pd.to_numeric(fact_txn['PATIENT_ID'], downcast='signed', errors='coerce').astype('Int64')
fact_txn['PHYSICIAN_ID'] = pd.to_numeric(fact_txn['PHYSICIAN_ID'], downcast='signed', errors='coerce').astype('Int64')
# Strings
fact_txn['PATIENT_ID'] = fact_txn['PATIENT_ID'].astype(str)  # Convert PATIENT_ID to string
fact_txn['PHYSICIAN_ID'] = fact_txn['PHYSICIAN_ID'].astype(str)  # Convert PHYSICIAN_ID to string
# Categories
fact_txn['TXN_LOCATION_TYPE'] = fact_txn['TXN_LOCATION_TYPE'].astype('category')
fact_txn['INSURANCE_TYPE'] = fact_txn['INSURANCE_TYPE'].astype('category')
fact_txn['TXN_TYPE'] = fact_txn['TXN_TYPE'].astype('category')
fact_txn['TXN_DESC'] = fact_txn['TXN_DESC'].astype('category')

# Rename columns
fact_txn.rename(columns={'PHYSICIAN_ID': 'HCP_ID'}, inplace=True)

# Display the cleaned and transformed dataframe
fact_txn.info()
fact_txn.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 115274 entries, 0 to 115273
Data columns (total 7 columns):
 #   Column             Non-Null Count   Dtype         
---  ------             --------------   -----         
 0   TXN_DT             115274 non-null  datetime64[ns]
 1   PATIENT_ID         115274 non-null  object        
 2   HCP_ID             115274 non-null  object        
 3   TXN_LOCATION_TYPE  99479 non-null   category      
 4   INSURANCE_TYPE     115274 non-null  category      
 5   TXN_TYPE           115274 non-null  category      
 6   TXN_DESC           115274 non-null  category      
dtypes: category(4), datetime64[ns](1), object(2)
memory usage: 3.1+ MB


TXN_DT PATIENT_ID HCP_ID       TXN_LOCATION_TYPE INSURANCE_TYPE  \
0 2021-12-27       3124   <NA>  INDEPENDENT LABORATORY     COMMERCIAL   
1 2020-02-25       3352  21972                  OFFICE     COMMERCIAL   
2 2022-01-25       2902  21586     HOSPITAL OUTPATIENT       MEDICARE   
3 2022-03-05       2943   2439              UNASSIGNED     COMMERCIAL   
4 2018-01-03       3589   6381                  OFFICE       MEDICARE   

     TXN_TYPE           TXN_DESC  
0    SYMPTOMS  ACUTE_PHARYNGITIS  
1  CONDITIONS            obesity  
2  CONDITIONS  immunocompromised  
3  CONDITIONS           diabetes  
4    SYMPTOMS              COUGH

In [160]:
# Apply upper case to TXN_DESC column to avoid missmatches 
fact_txn['TXN_DESC'] = fact_txn ['TXN_DESC'].str.upper()
fact_txn.head()

TXN_DT PATIENT_ID HCP_ID       TXN_LOCATION_TYPE INSURANCE_TYPE  \
0 2021-12-27       3124   <NA>  INDEPENDENT LABORATORY     COMMERCIAL   
1 2020-02-25       3352  21972                  OFFICE     COMMERCIAL   
2 2022-01-25       2902  21586     HOSPITAL OUTPATIENT       MEDICARE   
3 2022-03-05       2943   2439              UNASSIGNED     COMMERCIAL   
4 2018-01-03       3589   6381                  OFFICE       MEDICARE   

     TXN_TYPE           TXN_DESC  
0    SYMPTOMS  ACUTE_PHARYNGITIS  
1  CONDITIONS            OBESITY  
2  CONDITIONS  IMMUNOCOMPROMISED  
3  CONDITIONS           DIABETES  
4    SYMPTOMS              COUGH

In [161]:
# Clean and transform dim_patient dataframe
dim_patient = dataframes['dim_patient']

# Convert data types
# Numeric
dim_patient['PATIENT_ID'] = pd.to_numeric(dim_patient['PATIENT_ID'], downcast='signed', errors='coerce').astype('Int64')
dim_patient['BIRTH_YEAR'] = pd.to_numeric(dim_patient['BIRTH_YEAR'], downcast='signed', errors='coerce').astype('Int64')

# Strings
dim_patient['PATIENT_ID'] = dim_patient['PATIENT_ID'].astype(str)  # Convert PATIENT_ID to string

# Categories
dim_patient['GENDER'] = dim_patient['GENDER'].astype('category')

# Rename columns
dim_patient.rename(columns={'BIRTH_DATE': 'BIRTH_DT'}, inplace=True)


# Display the cleaned and transformed dataframe
dim_patient.info()
dim_patient.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4020 entries, 0 to 4019
Data columns (total 3 columns):
 #   Column      Non-Null Count  Dtype   
---  ------      --------------  -----   
 0   PATIENT_ID  4020 non-null   object  
 1   BIRTH_YEAR  4020 non-null   Int64   
 2   GENDER      4020 non-null   category
dtypes: Int64(1), category(1), object(1)
memory usage: 70.9+ KB


PATIENT_ID  BIRTH_YEAR GENDER
0          1        1988      M
1          2        2020      M
2          3        1973      M
3          4        2022      M
4          5        1988      M

In [162]:
# Clean and transform dim_physician dataframe
dim_physician = dataframes['dim_physician']

# Handle missing values if necessary (e.g., fill with a placeholder or drop)
# dim_physician['GENDER'].fillna('Unknown', inplace=True)
# dim_physician['BIRTH_YEAR'].fillna(-1, inplace=True)

# Convert data types
# Numeric
dim_physician['PHYSICIAN_ID'] = pd.to_numeric(dim_physician['PHYSICIAN_ID'], downcast='signed', errors='coerce').astype('Int64')
dim_physician['BIRTH_YEAR'] = pd.to_numeric(dim_physician['BIRTH_YEAR'], downcast='signed', errors='coerce').astype('Int64')

# Strings
dim_physician['PHYSICIAN_ID'] = dim_physician['PHYSICIAN_ID'].astype(str)  # Convert PHYSICIAN_ID to string

# Categories
dim_physician['STATE'] = dim_physician['STATE'].astype('category')
dim_physician['PHYSICIAN_TYPE'] = dim_physician['PHYSICIAN_TYPE'].astype('category')
dim_physician['GENDER'] = dim_physician['GENDER'].astype('category')

# Rename columns
dim_physician.rename(columns={'PHYSICIAN_TYPE': 'HCP_SPECIALTY'}, inplace=True)
dim_physician.rename(columns={'PHYSICIAN_ID': 'HCP_ID'}, inplace=True)
dim_physician.rename(columns={'BIRTH_DATE': 'BIRTH_DT'}, inplace=True)

# Display the cleaned and transformed dataframe
dim_physician.info()
dim_physician.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 25343 entries, 0 to 25342
Data columns (total 5 columns):
 #   Column         Non-Null Count  Dtype   
---  ------         --------------  -----   
 0   HCP_ID         25343 non-null  object  
 1   STATE          25341 non-null  category
 2   HCP_SPECIALTY  25343 non-null  category
 3   GENDER         24137 non-null  category
 4   BIRTH_YEAR     21039 non-null  Int64   
dtypes: Int64(1), category(3), object(1)
memory usage: 532.1+ KB


HCP_ID STATE        HCP_SPECIALTY GENDER  BIRTH_YEAR
0      1    TX     PHYSICAL THERAPY    NaN        <NA>
1      2    IN  PHYSICIAN ASSISTANT    NaN        <NA>
2      3    CA   EMERGENCY MEDICINE    NaN        <NA>
3      4    TX   NURSE PRACTITIONER    NaN        <NA>
4      5    WA   EMERGENCY MEDICINE    NaN        <NA>

In [163]:
# Age calculator function 
from datetime import date
def calculate_age(birth):
    today = date.today()
    return today.year - birth.year - ((today.month, today.day) < (birth.month, birth.day))

In [164]:
dim_physician['BIRTH_DT'] = pd.to_datetime(dim_physician['BIRTH_YEAR'], format = '%Y')
dim_patient['BIRTH_DT'] = pd.to_datetime(dim_patient['BIRTH_YEAR'], format = '%Y')

In [165]:
# Calculate current age (aprox.) of the patient & hcp as a new column
dim_patient['PATIENT_AGE'] = dim_patient['BIRTH_DT'].apply(calculate_age)
dim_patient.head()


PATIENT_ID  BIRTH_YEAR GENDER   BIRTH_DT  PATIENT_AGE
0          1        1988      M 1988-01-01           36
1          2        2020      M 2020-01-01            4
2          3        1973      M 1973-01-01           51
3          4        2022      M 2022-01-01            2
4          5        1988      M 1988-01-01           36

In [166]:
# Calculate current age (aprox.) of the patient & hcp as a new column
dim_physician['HCP_AGE'] = dim_physician['BIRTH_DT'].apply(calculate_age)
dim_physician['HCP_AGE'] = pd.to_numeric(dim_physician['HCP_AGE'], downcast='signed', errors='coerce').astype('Int64')
dim_physician.head()

HCP_ID STATE        HCP_SPECIALTY GENDER  BIRTH_YEAR BIRTH_DT  HCP_AGE
0      1    TX     PHYSICAL THERAPY    NaN        <NA>      NaT     <NA>
1      2    IN  PHYSICIAN ASSISTANT    NaN        <NA>      NaT     <NA>
2      3    CA   EMERGENCY MEDICINE    NaN        <NA>      NaT     <NA>
3      4    TX   NURSE PRACTITIONER    NaN        <NA>      NaT     <NA>
4      5    WA   EMERGENCY MEDICINE    NaN        <NA>      NaT     <NA>

In [167]:
# Format PATIENT_GENDER as expected
def gender_formater(value):
    if value == 'M':
        return 'M-Male'
    elif value == 'F':
        return 'F-Female'
    else:
        return 'U-Unknown'

In [168]:
dim_patient['PATIENT_GENDER'] = dim_patient['GENDER'].map(gender_formater)
dim_patient.head(10)

PATIENT_ID  BIRTH_YEAR GENDER   BIRTH_DT  PATIENT_AGE PATIENT_GENDER
0          1        1988      M 1988-01-01           36         M-Male
1          2        2020      M 2020-01-01            4         M-Male
2          3        1973      M 1973-01-01           51         M-Male
3          4        2022      M 2022-01-01            2         M-Male
4          5        1988      M 1988-01-01           36         M-Male
5          6        1951      M 1951-01-01           73         M-Male
6          7        2021      M 2021-01-01            3         M-Male
7          8        1977      M 1977-01-01           47         M-Male
8          9        1991      M 1991-01-01           33         M-Male
9         10        1948      F 1948-01-01           76       F-Female

In [169]:
dim_physician['HCP_GENDER'] = dim_physician['GENDER'].map(gender_formater)
dim_physician.head(10)

HCP_ID STATE        HCP_SPECIALTY GENDER  BIRTH_YEAR BIRTH_DT  HCP_AGE  \
0      1    TX     PHYSICAL THERAPY    NaN        <NA>      NaT     <NA>   
1      2    IN  PHYSICIAN ASSISTANT    NaN        <NA>      NaT     <NA>   
2      3    CA   EMERGENCY MEDICINE    NaN        <NA>      NaT     <NA>   
3      4    TX   NURSE PRACTITIONER    NaN        <NA>      NaT     <NA>   
4      5    WA   EMERGENCY MEDICINE    NaN        <NA>      NaT     <NA>   
5      6    OH   NURSE PRACTITIONER    NaN        <NA>      NaT     <NA>   
6      7    MS   NURSE PRACTITIONER    NaN        <NA>      NaT     <NA>   
7      8    CA      FAMILY MEDICINE    NaN        <NA>      NaT     <NA>   
8      9    SC  PHYSICIAN ASSISTANT    NaN        <NA>      NaT     <NA>   
9     10    NJ   NURSE PRACTITIONER    NaN        <NA>      NaT     <NA>   

  HCP_GENDER  
0  U-Unknown  
1  U-Unknown  
2  U-Unknown  
3  U-Unknown  
4  U-Unknown  
5  U-Unknown  
6  U-Unknown  
7  U-Unknown  
8  U-Unknown  
9  U-Unknown

In [170]:
# Get state name from abbreviation from dim_physician
import us 
state_dict = us.states.mapping('abbr', 'name')
dim_physician['STATE_NAME'] = dim_physician['STATE'].map(state_dict)
dim_physician['STATE_NAME'] = dim_physician['STATE_NAME'].astype('category')
dim_physician.head()

HCP_ID STATE        HCP_SPECIALTY GENDER  BIRTH_YEAR BIRTH_DT  HCP_AGE  \
0      1    TX     PHYSICAL THERAPY    NaN        <NA>      NaT     <NA>   
1      2    IN  PHYSICIAN ASSISTANT    NaN        <NA>      NaT     <NA>   
2      3    CA   EMERGENCY MEDICINE    NaN        <NA>      NaT     <NA>   
3      4    TX   NURSE PRACTITIONER    NaN        <NA>      NaT     <NA>   
4      5    WA   EMERGENCY MEDICINE    NaN        <NA>      NaT     <NA>   

  HCP_GENDER  STATE_NAME  
0  U-Unknown       Texas  
1  U-Unknown     Indiana  
2  U-Unknown  California  
3  U-Unknown       Texas  
4  U-Unknown  Washington

In [ ]:
dim_physician.drop(columns=['BIRTH_DT'], inplace=True)
#dim_patient.drop(columns=['BIRTH_DT'], inplace=True)
dim_physician.drop(columns=['BIRTH_YEAR'], inplace=True)
dim_patient.drop(columns=['BIRTH_YEAR'], inplace=True)
dim_physician.drop(columns=['GENDER'], inplace=True)
dim_patient.drop(columns=['GENDER'], inplace=True)


In [172]:
# Define the directory to save the intermediate CSV files
intermediate_dir = '../../data/intermediate'

# Ensure the directory exists
os.makedirs(intermediate_dir, exist_ok=True)

# Save each dataframe to a CSV file in the intermediate directory
for name, df in dataframes.items():
    file_path = os.path.join(intermediate_dir, f"{name}.csv")
    df.to_csv(file_path, index=False)
    print(f"Saved {name} to {file_path}")

Saved dim_physician to ../../data/intermediate/dim_physician.csv
Saved model_table to ../../data/intermediate/model_table.csv
Saved fact_txn to ../../data/intermediate/fact_txn.csv
Saved dim_patient to ../../data/intermediate/dim_patient.csv
Saved data_dic to ../../data/intermediate/data_dic.csv
